.github/workflows/ci_tests.yaml

```diff
@@ -0,0 +1,31 @@
+name: Python package
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    paths-ignore:
+      - 'docs/**'
+      - '**.md'
+
+jobs:
+
+  pytest_tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.9"]
+    container:
+      image: python:${{ matrix.python-version }}
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8 pytest pytest-cov pytest-xdist numpy
+          if [ -f requirements.txt ]; then pip install -r requirements.txt --find-links https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html; fi
+      - name: Test with pytest
+        run: |
+          pytest --cov=spmd tests/
```

Comment:
You might want to use the trick in https://github.com/wanchaol/spmd/pull/2 to add the newline, or configure your editor. :-)